site/src/app/filters.ts

```diff
@@ -7,39 +7,60 @@ import Fuse from 'fuse.js';
 
 import { NormalisedApp } from '../schema';
 
-export interface Filters {
-    textSearch: string;
+
+function filterAppName(apps: NormalisedApp[], search: string): NormalisedApp[] {
+    const fuse = new Fuse(apps, { keys: ['owner.name', 'title', 'description', 'name', 'tags'] });
+    const results = fuse.search(search);
+    return results.map((result) => result.item);
+}
+
+function filterNcsVersion(apps: NormalisedApp[], search: string): NormalisedApp[] {
+    const exactMatches: NormalisedApp[] = [];
+
+    for (const app of apps) {
+        for (const ncs of app.compatibleNcs) {
+            if (ncs.includes(search)) {
+                exactMatches.push(app);
+                break;
+            }
+        }
+    }
+
+    return exactMatches;
```

Comment:
Instead of pushing the found matched into an array, make it more functional:
```suggestion
    return apps.filter((app) => app.compatibleNcs.some((ncs) => ncs.includes(search)));
```